crates/wasi-nn/src/openvino.rs

```diff
@@ -40,7 +40,9 @@ impl Backend for OpenvinoBackend {
             .0
             .as_mut()
             .expect("openvino::Core was previously constructed");
-        let cnn_network = core.read_network_from_buffer(&xml, &weights)?;
+        let mut cnn_network = core.read_network_from_buffer(&xml, &weights)?;
+        let input_name = &cnn_network.get_input_name(0)?;
+        cnn_network.set_input_layout(input_name, Layout::NHWC)?;
```

Comment:
Instead of hard-coding this here, can we do this down in `set_input` by the TODO item at line 78 (with the appropriate input ID from the `index` argument)? There is a need to figure out how to allow different kinds of image layouts but for now keeping it at NHWC should be fine, especially if we keep it close to the TODO below.

Reply:
This would require changing Arc<openvino::CNNNetwork> to Arc<Mutex<openvino::CNNNetwork>> in order to get a mutable copy of the CNNNetwork, which is needed for set_input_layout. Even doing this I'm currently still getting the parameter mismatch error moving it here. I'll try a few more things and see about removing the hard coded value.